src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java

```diff
@@ -198,6 +198,39 @@ void createBuildPlanForExercise(ProgrammingExercise exercise, String planKey, Vc
     void updatePlanRepository(String buildProjectKey, String buildPlanKey, String ciRepoName, String repoProjectKey, String newRepoUrl, String existingRepoUrl, String newBranch,
             Optional<List<String>> optionalTriggeredByRepositories);
 
+    /**
+     * Returns a list of all notification ids for the given build plan for this server
+     *
+     * @param buildPlanKey The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @return a list of all notification ids
+     */
+    List<Long> getAllArtemisBuildPlanServerNotificationIds(String buildPlanKey);
+
+    /**
+     * Deletes the given notification id for the given build plan
+     *
+     * @param buildPlanKey         The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @param serverNotificationId the id of the notification to delete
+     */
+    void deleteBuildPlanServerNotificationId(String buildPlanKey, Long serverNotificationId);
+
+    /**
+     * Creates a new notification for the given build plan
+     *
+     * @param buildPlanKey          The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @param serverNotificationUrl the url of the endpoint to notify
+     */
+    void createBuildPlanServerNotification(String buildPlanKey, String serverNotificationUrl);
+
+    /**
+     * Reads all notifications on the current build plan and updates old notifications to the current endpoint
+     *
+     * @param projectKey    The key of the project, e.g. 'EIST16W1', which is normally the programming exercise project key.
+     * @param buildPlanKey  The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @param repositoryUrl the URL of the assignment repository
+     */
+    void fixBuildPlanNotification(String projectKey, String buildPlanKey, VcsRepositoryUrl repositoryUrl);
```

Comment:
Maybe a better name would be `updateBuildPlanNotificationUrl`?

Reply:
I decided on `overrideBuildPlanNotificationUrl` because with `update`, I would expect some parameter.